geps/gep-2627/index.md

```diff
@@ -0,0 +1,50 @@
+
+## Goals
+* Provide a way for Gateway API resource owners to mark their resources as relevant for external DNS provisioning
+* Ensure that the above method has a way for multiple providers to be present in the cluster and be able to actuate external DNS provisioning requests
+* Ensure that any method is based on structured fields and makes the most of `status` on whatever resources are relevant, whether they are existing Gateway API resources or new resources.
+* Increase portability and supportability between Gateway API implementations and third party controllers offering DNS integration.
+* Clarity on the scope of hostnames under management. (This should _not_ be able to be used to affect the standard in-cluster DNS configuration)
+
+## Non-Goals
+
+* Anything to do with configuring in-cluster DNS. This support is for configuration outside the cluster only.
+* Ways to define if the Gateway API resources are allowed to request particular hostnames. These choices should be left to the implementations that actually actuate the requests for hostnames. However, `status` flows should be specified so as to make clear if a hostname provisioning request cannot be performed.
+* Cover more complex DNS routing strategies that come into play for multi-cluster topologies such as round robin, failover, health checks, weighted and geo location with this first pass. Supporting these types of use cases for distributed gateways (e.g., in different regions or multiple gateways for resilience within a region) and offering a form of global load balancing leveraging DNS is a potential future goal.
+
+## Use Cases
+
+As a cluster administrator, I manage a set of domains and a set of gateways. I would like to declaratively define which gateways should be used for provisioning DNS records, and, if necessary, which DNS provider to use to configure connectivity for clients accessing these domains and my gateway so that I can see and configure which DNS provider is being used.
+
+As a cluster administrator, I would like to have the DNS names automatically populated into my specified DNS zones as a set of records based on the assigned addresses of my gateways so that I do not have to undertake external automation or management of this essential task.
+
+As a cluster administrator I would have the status of the DNS records reported back to me, so that I can leverage existing kube based monitoring tools to know the status of the integration.
+
+As a cluster administrator, I would like the DNS records to be updated automatically if the `spec` of assigned gateways changes, whether those changes are for IP address or hostname. 
+As a DNS administrator, I should be able to ensure that only approved External DNS controllers can make changes to DNS zone configuration. (This should in general be taken care of by DNS system <-> External DNS controller interactions like user credentials and operation status responses, but it is important to remember that it needs to happen).
```

Comment:
We don't cover any application developer user stories here, we seem pretty focused on the cluster administrator. I anticipate that application developers will in fact will have relevant use cases. What are your thoughts on application developer roles needs here?

Reply:
I am struggling a little to think of specific application developer use cases around this. What comes to mind is allowing the DNSPolicy to potentially target a HTTPRoute or possibly some form of health check based on the available endpoints behind a service. So as an attempt at a use case "As an application developer, I do not want my service to be exposed via the Gateway until there are pods that are healthy and ready to serve the traffic" ? Did you have anything specific in mind?

Reply:
will mark as resolved as we haven't had any input here. Things like a default health check based on endpoints are nice to haves but not integral.